### PR TITLE
UINV-487 "Account number" value must be `null` for the default value of "Accounting code"

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://issues.folio.org/browse/UINV-487
https://issues.folio.org/browse/UINV-227 regression

The hyphen (`-`) must be displayed as the account number value on the voucher details page if a default accounting code has been selected for the invoice. The batch voucher export file contains only populated values `accountNo` fields.

## Approach

The "Accounting code" field's options are based on vendor accounts (an option for each account with a defined "accounting code" value). If a vendor has an ERP code it serves as the additional default option for the accounting code and at the same time value `accountNo` field must be `null`.

For the correct display of the selected option in the "Accounting code" field, for the default option, the account number value is set to `NO_ACCOUNT_NUMBER` as a temporary value. When submitting the form, this value is replaced with `null`.

<details open><summary>Details</summary>
    <h4>Default option (accounting code == ERP code, account number == null)</h4>

chrome_decuQveH6c

---

 <h4>Account's option (accounting code == vendor account's accounting code (`appSystemNo`), account number == vendor account's account number)</h4>
    <p>

chrome_h7P99HG11D

---

<h4>Empty "Accounting code" field</h4>

chrome_yYQcEC6m2K

---

<h4>Batch voucher export report</h4>

chrome_KKoAIBCSBJ

</details> 

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by usavkov-epam** [comment]
Alternative for https://github.com/folio-org/ui-invoice/pull/703

**Review by zburke — COMMENTED** [review]
This code would really benefit from some comments describing the relationship between account number and account code, and when an account number is supposed to receive an empty value vs a default value. There are some comments in the tests, but that's it, and it's really hard to assess the correctness of any related code. The Jira ticket describes a particular instance of incorrect behavior, but without a clear description of _correct_ behavior I don't feel comfortable approving this. 

Side note, the `getAccountingCodeOptions` function is English-only (`Default` is hard-coded). I filed [UINV-489](https://issues.folio.org/browse/UINV-489). As a rule, all text must be pushed through the i18n layer via `<FormattedMessage>` or `useIntl.formatMessage()`. If you notice other hard-coded strings in this application, please file tickets to fix them.

File: src/common/utils/getAccountingCodeOptions/getAccountingCodeOptions.js [hunk 1]
```diff
@@ -9,7 +11,7 @@ export const getAccountingCodeOptions = (vendor) => {
   }));
   const erpCode = get(vendor, 'erpCode');
   const defaultOption = erpCode
-    ? [{ label: `Default (${erpCode})`, value: erpCode }]
+    ? [{ label: `Default (${erpCode})`, value: NO_ACCOUNT_NUMBER }]
```
[the remark this answers is not in the document]
Comment (usavkov-epam) [reply src/common/utils/getAccountingCodeOptions/getAccountingCodeOptions.js]: Thanks for the review, Zak!
Yeah, agree about label translation, it definitely makes sense to localize it.

Basically, the option's label looks like `"Account-No (Accounting-Code)"`, and for the default option, the value of the accounting code would be `erpCode`.

I added some comments on how these options are build.

File: src/invoices/InvoiceForm/InvoiceForm.js [hunk 2]
```diff
@@ -215,7 +218,7 @@ const InvoiceForm = ({
   );
 
   const onChangeAccNumber = useCallback(accNumber => {
-    const accCode = accNumber !== invoiceVendor.erpCode
+    const accCode = accNumber !== NO_ACCOUNT_NUMBER
```
[the remark this answers is not in the document]
Comment (usavkov-epam) [reply src/invoices/InvoiceForm/InvoiceForm.js:226]: I added a description for the "Accounting code" field options builder (`getAccountingCodeOptions`) and also add a comment on the reason for using `NO_ACCOUNT_NUMBER`.

**Review by zburke — APPROVED** [review]

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**